pennylane/estimator/templates/select.py

```diff
@@ -43,6 +43,10 @@ class SelectTHC(ResourceOperator):
     Args:
         thc_ham (:class:`~pennylane.estimator.compact_hamiltonian.THCHamiltonian`): A tensor hypercontracted
             Hamiltonian on which the select operator is being applied.
+        batched_rotations (int | None): The number of rotation angles to load simultaneously
+            into temporary quantum registers for processing in a batch of Givens rotation circuits.
+            This parameter manages the trade-off between qubit count and gate cost. Default is all angles
+            are loaded at once, resulting in the highest qubit cost but the lowest gate cost.
```

Comment:
Please try to make the first sentence more clear. This expects a significant amount of user knowledge.